### PR TITLE
Added searches, updated natrules, and fixed minor issues with hitcounts and apiclasstemplate

### PR DESCRIPTION
**Pull request by Niltak**
Added three searches:
  globalsearch
  object
  policy

Fixed issue with how searches work with the current 'get' in apiclasstemplate. Also fixed minor logging issue.

Updated natrules to take nat_id's as well as nat_name. Changed variable from name to nat_name similar to acp.

Added offset to hitcounts to be able to loop hitcounts to grab all rules. Removed pulling items out of the response to keep it consistent with the other classes. *If you want I can add a loop to collect all of the hit count records so that the user doesn't have to build out their own logic.

TODO:
Build out unit_tests

### CODE REVIEW

**Comment by Niltak** [comment]
Added unit tests

**Comment by marksull** [comment]
Wow, detailed pull request. Looks great. I'll give it a test on Monday...

File: fmcapi/api_objects/policy_services/natrules.py [hunk 1]
```diff
@@ -25,26 +26,43 @@ def __init__(self, fmc, **kwargs):
         logging.debug("In __init__() for NatRules class.")
         self.parse_kwargs(**kwargs)
 
-    def nat_policy(self, name):
+    def nat_policy(self, nat_name=None, nat_id=None):
         """
         Associate NAT Policy.
 
         :param name: (str) Name of NAT Policy.
+        :param id: (str) ID of NAT Policy.
```
Comment (marksull) [inline fmcapi/api_objects/policy_services/natrules.py:34]: Should be `nat_id` ;-)

File: fmcapi/api_objects/policy_services/natrules.py [hunk 2]
```diff
@@ -60,9 +60,11 @@ def parse_kwargs(self, **kwargs):
         super().parse_kwargs(**kwargs)
         logging.debug("In parse_kwargs() for NatRules class.")
         if "nat_id" in kwargs:
-            self.nat_policy(nat_id=kwargs["nat_id"])
+            self.nat_policy(id=kwargs["nat_id"])
         elif "nat_name" in kwargs:
-            self.nat_policy(nat_name=kwargs["nat_name"])
+            self.nat_policy(name=kwargs["nat_name"])
+        elif "name" in kwargs:
+            self.nat_policy(name=kwargs["name"])
```
Comment (Niltak) [inline fmcapi/api_objects/policy_services/natrules.py:67]: Added backwards compatibility and acp like variables.

File: fmcapi/api_objects/policy_services/natrules.py [hunk 3]
```diff
@@ -25,26 +26,43 @@ def __init__(self, fmc, **kwargs):
         logging.debug("In __init__() for NatRules class.")
         self.parse_kwargs(**kwargs)
 
-    def nat_policy(self, name):
+    def nat_policy(self, nat_name=None, nat_id=None):
```
[the remark this answers is not in the document]
Comment (Niltak) [reply fmcapi/api_objects/policy_services/natrules.py]: https://github.com/marksull/fmcapi/pull/127#discussion_r956358829

File: fmcapi/api_objects/policy_services/natrules.py (hunk 1 above)
Comment (Niltak) [reply fmcapi/api_objects/policy_services/natrules.py:34]: This should be corrected with the changed variable names for the nat_policy and lines up with acp in other classes.

**Pull request merged**